### PR TITLE
Add nightly build pipeline, remove imx8qm variant

### PR DESCRIPTION
**Pull request by mnokka-unikie**
Adding second pipeline (copy of current main build) for nightly operations, removed imx8qm variant (due main pipeline changes).

### CODE REVIEW

File: ghaf-nightly-build.groovy [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+pipeline {
+  agent any
+  parameters {
+    string name: 'URL', defaultValue: 'https://github.com/tiiuae/ghaf.git'
+    string name: 'BRANCH', defaultValue: 'main'
+  }
+  triggers {
+     cron('@midnight')
+  }
```
Comment (henrirosten) [inline ghaf-nightly-build.groovy:15]: If the nightly trigger is the only change compared to https://github.com/tiiuae/ghaf-jenkins-pipeline/blob/main/ghaf-build-pipeline.groovy, wouldn't it be enough to add the nightly trigger to https://github.com/tiiuae/ghaf-jenkins-pipeline/blob/main/ghaf-build-pipeline.groovy?

I mean, we don't need to introduce a new pipeline if we only want to change the triggers, right?

Comment (mnokka-unikie) [reply ghaf-nightly-build.groovy:15]: Orders from the boss....

Comment (mnokka-unikie) [reply ghaf-nightly-build.groovy:15]: Logically having this "triggering" in the original main would make sense. But as main will be removed from all extra (analyze, provenance) and nightly will do these extra activities,  then this move makes sense...

**Comment by henrirosten** [comment]
@mnokka-unikie : This PR needs to be merged before https://github.com/tiiuae/ghaf-infra/pull/151.

Also, @mnokka-unikie  commented offline that there will be a follow-up PR which will remove some steps from  the current main pipeline with an attempt to make it faster, effectively moving some of the steps from the main pipeline to the nightly pipeline @mnokka-unikie is introducing in this PR.

File: ghaf-nightly-build.groovy [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          sh 'nix run github:tiiuae/sbomnix/a1f0f88d719687acedd989899ecd7fafab42394c#vulnxscan -- .#packages.x86_64-linux.generic-x86_64-debug --out result-vulns-generic-x86_64-debug.csv'
+          sh 'nix run github:tiiuae/sbomnix/a1f0f88d719687acedd989899ecd7fafab42394c#vulnxscan -- .#packages.x86_64-linux.lenovo-x1-carbon-gen11-debug --out result-vulns-lenovo-x1-carbon-gen11-debug.csv'
+          sh 'nix run github:tiiuae/sbomnix/a1f0f88d719687acedd989899ecd7fafab42394c#vulnxscan -- .#packages.riscv64-linux.microchip-icicle-kit-debug --out result-vulns-microchip-icicle-kit-debug.csv'
+          sh 'nix run github:tiiuae/sbomnix/a1f0f88d719687acedd989899ecd7fafab42394c#vulnxscan -- .#packages.aarch64-linux.nvidia-jetson-orin-agx-debug --out result-vulns-aarch64-jetson-orin-agx-debug.csv'
+          sh 'nix run github:tiiuae/sbomnix/a1f0f88d719687acedd989899ecd7fafab42394c#vulnxscan -- .#packages.aarch64-linux.nvidia-jetson-orin-nx-debug --out result-vulns-aarch64-jetson-orin-nx-debug.csv'
+      }
```
Comment (henrirosten) [inline ghaf-nightly-build.groovy:125]: Missing one closing curly bracket above this one. 
Has this pipeline been tested, or does Jenkins simply ignore the missing bracket here?

Comment (mnokka-unikie) [reply ghaf-nightly-build.groovy:125]: System has been running 2 nights every 30 mins... But stage section do need closure, some last minute hassle...I will add and push again

**Review by henrirosten — APPROVED** [review]

**Pull request merged**